<a href="https://colab.research.google.com/github/kimsoohwa/jejumaster/blob/main/1%EC%9D%BC%EC%B0%A8_%ED%8C%8C%EC%9D%B4%EC%8D%AC%EA%B8%B0%EC%B4%88%EB%8B%A4%EC%A7%80%EA%B8%B0_%EC%8B%A4%EC%8A%B52.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 1일차-파이썬 기초 다지기

## 1.파이썬 이해하기

### 1-2.파이썬 학습환경 만들기

#### 파이썬 설치 위치 확인

In [ ]:
import sys

sys.executable

In [ ]:
print(sys.path)

### 1-3.파이썬 라이브러리 사용 방법

#### 설치된 라이브러리 전체 확인

In [ ]:
pip list

#### 특정 라이브러리 설치 확인

In [ ]:
pip show numpy

#### 라이브러리 설치 및 삭제

In [ ]:
# 라이브러리 설치
pip install numpy

# 라이브러리 삭제
# pip uninstall numpy

### 수업에 사용할 라이브러리 일부 설치하기
(PC 윈도우 명령프롬프트에서 먼저 실행) pip를 최신으로 업그레이드한 후
파이썬 라이브러리를 설치하도록 한다.
- python -m pip install --upgrade pip  

In [ ]:
!pip install numpy

### 1-4.파이썬 기초 입문하기

#### 파이썬 오류&예외처리

In [ ]:
# Syntax 오류 발생 예
a = 1
if a < 9
    print('a < 9')

SyntaxError: ignored

In [ ]:
# 오류 예외처리
while True:
    try:
        x = int(input('Please enter a number: '))
        break

    except ValueError:
        print('Oops! That was no valid number. Try again...')

print('-----')

Please enter a number: 10
-----


### [실습] 파이썬 정보 확인하기


1.   설치된 파이썬 버전 출력하기
2.   파이썬 실행파일 위치 출력하기



In [ ]:
#버젼
import sys
print(f'파이썬 버젼 :{sys.version}')
#위치
print(sys.path)
print(f'파이썬 위치 :{sys.executable}')

파이썬 버젼 :3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0]
['/content', '/env/python', '/usr/lib/python310.zip', '/usr/lib/python3.10', '/usr/lib/python3.10/lib-dynload', '', '/usr/local/lib/python3.10/dist-packages', '/usr/lib/python3/dist-packages', '/usr/local/lib/python3.10/dist-packages/IPython/extensions', '/root/.ipython']
파이썬 위치 :/usr/bin/python3


----------

## 2.파이썬 기초 다지기

### 2-1.파이썬 자료형과 연산자

### **[파이썬 자료형]**

### [실습문제]


-  문제1: 10 + 3 x 4 결과 출력
- 문제2: 아이폰 15대를 가지고 있다. 아이폰의 가격이 130만원 일 때 총 금액은?
- 문제3: 주어진 자연수가 홀수인지 짝수인지 판별하는 방법은?
- 문제4: 2x2x2x5x5x5x5x3x3x3x3을 제곱형태로 나타낸 후 그 결과 출력하기



In [ ]:
result = 10+3*4
print(result)




22


#### 파이썬 자료형 확인

In [ ]:
d1 = 123
d2 = '123'
d3 = [123]
d4 = (123,)
d5 = {'key':123}
d6 = {123}
d7 = True

print(f'{d1}의 자료형: {type(d1)}')
print(f'{d2}의 자료형: {type(d2)}')
print(f'{d3}의 자료형: {type(d3)}')
print(f'{d4}의 자료형: {type(d4)}')
print(f'{d5}의 자료형: {type(d5)}')
print(f'{d6}의 자료형: {type(d6)}')
print(f'{d7}의 자료형: {type(d7)}')

123의 자료형: <class 'int'>
123의 자료형: <class 'str'>
[123]의 자료형: <class 'list'>
(123,)의 자료형: <class 'tuple'>
{'key': 123}의 자료형: <class 'dict'>
{123}의 자료형: <class 'set'>
True의 자료형: <class 'bool'>


In [ ]:
datas = [123,'123',[123],(123,),{'key':123},{123},True]

for d in datas:
         print(f"데이터 '{d}' 의 자료형: {type(d)}")

데이터 '123' 의 자료형: <class 'int'>
데이터 '123' 의 자료형: <class 'str'>
데이터 '[123]' 의 자료형: <class 'list'>
데이터 '(123,)' 의 자료형: <class 'tuple'>
데이터 '{'key': 123}' 의 자료형: <class 'dict'>
데이터 '{123}' 의 자료형: <class 'set'>
데이터 'True' 의 자료형: <class 'bool'>


#### 실습문제
두 수를 입력 받아 덧셈, 뺄셈, 곱셈, 몫, 나머지를 출력하는 프로그램을 작성하세요


In [ ]:
num1= int(input('첫번째 숫자를 입력하세요.'))
num2= int(input('두번째 숫자를 입력하세요.'))
print('='*50)

plus =num1+num2
print(f'덧셈:{num1}+{num2}={plus}')

minus =num1-num2
print(f'뺄셈:{num1}-{num2}={minus}')

multip=num1*num2
print(f'곱셈:{num1}*{num2}={multip}')

share=num1//num2
print(f'몫:{num1}//{num2}={share}')

rest=num1%num2
print(f'나머지:{num1}%{num2}={rest}')



첫번째 숫자를 입력하세요.50
두번째 숫자를 입력하세요.3
덧셈:50+3=53
뺄셈:50-3=47
곱셈:50*3=150
몫:50//3=16
나머지:50%3=2


In [ ]:
num1 = int(input('첫번째 숫자 :'))
num2 = int(input('두번째 숫자 :'))
print('*'*50)

print(f' 뎃셈: {num1}+{num2}={num1+num2}')
print(f' 뺄셈: {num1}-{num2}={num1-num2}')
print(f' 곱셈: {num1}*{num2}={num1*num2}')
print(f' 몫: {num1}//{num2}={num1//num2}')
print(f' 나머지: {num1}%{num2}={num1%num2}')

첫번째 숫자 :50
두번째 숫자 :3
**************************************************
 뎃셈: 50+3=53
 뺄셈: 50-3=47
 곱셈: 50*3=150
 몫: 50//3=16
 나머지: 50%3=2


#### 실습문제 :  시간 계산 프로그램
속력과 거리를 입력 받아 시간을 계산하여 출력하는 프로그램을 만드세요!
- 시간 = 거리 / 속력



In [ ]:
velocity = input('속력를 입력해 주세요(km/h) : ')
distance = input('거리를 입력해 주세요(km) : ')

velocity = int(input('속력을 입력해 주세요(km/h):'))
distance =int(input('거리를 입력해 두세요(km):'))
print('-'*50)
print(f"속력: {velocity} km/h ")
print(f"거리: {distance} km ")
time=distance/velocity
print(f"시간: {time} h")



- eval()함수 사용하여 문자열을 계산할 수 있다

In [ ]:
# eval()함수 사용
time = eval(f'{distance} / {velocity}')
print(time)

3.4


#### 실습문제 : 정수 합 구하기
아래와 같이 정수 n을 입력 받아, a + aa + aaa의 결과 값을 구하는 프로그램을 작성하세요

In [ ]:

a=input('정수를 입력하세요:')
a=int(a)
aa=int(f'{a}{a}')
aaa=int(f'{a}{a}{a}')
result=a+aa+aaa
print(f'{a}+{aa}+{aaa}={result}')

정수를 입력하세요:7
7+77+777=861


In [ ]:
a = input('숫자 입력: ')
aaa = int(a+a+a)
aa = int(a+a)
a = int(a)
print(f'{a} + {aa} + {aaa} = {a+aa+aaa}')


숫자 입력: 7
7 + 77 + 777 = 861


In [ ]:
a = input('숫자 하나를 입력하세요: ')
print(f"{a}+{aa}+{aaa} = {eval(f'{a}+{aa}+{aaa}')}")
# value = f'{a}+{aa}+{aaa}'
# print(f"{value} = {eval(value)}")

숫자 하나를 입력하세요: 7
7+77+777 = 861


#### 내장 수학 함수
- 파이썬 내장 함수 https://docs.python.org/ko/3.10/library/functions.html

In [ ]:
import math

print(f'원주율: {math.pi}')
print(f'절대값: {abs(-10)}')
print(f'반올림: {round(1.791, 2)}')
print(f'올림  : {math.ceil(1.7)}')
print(f'내림  : {math.floor(1.7)}')
print(f'제곱  : {math.pow(2, 10)}')
print(f'제곱근: {math.sqrt(4)}')
print(f'로그  : {math.log(10) }')

원주율: 3.141592653589793
절대값: 10
반올림: 1.79
올림  : 2
내림  : 1
제곱  : 1024.0
제곱근: 2.0
로그  : 2.302585092994046


### [실습문제] : 수학 함수 사용하기
- 문제1: 반지름이 4인 원의 넓이 출력
- 문제2: 3.14 곱하가 2 나누기 7의 값을 반올림하여 계산 결과 출력
- 문제3: 원주율을 3.14로 만들어서 반지름이 4인 원의 둘레의 길이 출력

-----

### **[문자와 문자열]**

#### 문자와 문자열 변환

In [ ]:
# 문자 --> 숫자(정수) 로 변환 : ord() 함수 사용
print( ord('A') )
print( ord('!') )
print( ord('가') )

65
33
44032


In [ ]:
# 숫자(정수) --> 문자 로 변환 : chr() 함수 사용
print( chr(65) )
print( chr(66) )
print( chr(97) )
print( chr(98) )
print( chr(44032) )
print( chr(44033) )

A
B
a
b
가
각


In [ ]:
# 문자 --> 숫자 --> 이진수 로 변환
for c in '파이썬':
    숫자변환 = ord(c)              # unicode
    이진수변환 = bin(숫자변환)     # 16진수 binarycode
    print( 이진수변환, end = ' ')  # 한 줄로 표현하기

In [ ]:
for c in '파이썬':
    # 이진수에서 앞'0b'를 없애기 위해 format() 함수 사용할 수 있다.
    print( format(ord(c), 'b'), end = ' ')  # 한 줄로 표현하기

#### 문자열 따옴표로 지정

In [ ]:
# 큰 따옴표 or 작은 따옴표로 지정한다
print('Hello')
print("Hello")
print("Hello 'world")
print('Hello "world')
print(' "Yes, I can!" ')

In [ ]:
#따옴표 세 개 사용 : 여러 줄로 표현할 때
a = '''박씨가 소리질렀다.
도둑이야!'''
print(a)

In [ ]:
menu = '''OOO카페 메뉴판
- 아메리카노:  3000원
- 카페라테:     4000원
- 카페모카:     5000원'''
print(menu)
coffee = input('주문하실 음료를 입력하세요(ex:아메리카노) => ')
cup = input(f'{coffee} 몇 잔을 주문하시겠습니까?: ')

print(f'주문하신 음료는 {coffee} {cup} 잔 입니다.')

OOO카페 메뉴판
- 아메리카노:  3000원
- 카페라테:     4000원
- 카페모카:     5000원
주문하실 음료를 입력하세요(ex:아메리카노) => 아메리카노
아메리카노 몇 잔을 주문하시겠습니까?: 3
주문하신 음료는 아메리카노 3 잔 입니다.


#### 문자열 연산
- (+): 문자열 연결하기
- (*) : 반복의 의미

In [ ]:
# 문자열 연결하기
print('Hello'+'world')
print('Hello','world', '!')

a = "Python"
b = " is very fun!"
print(a+b)

In [ ]:
# 반복의 의미
print('Hello'*3)
print('-' * 30)

### 실습문제 : 문자열 출력하기
- 문제1: 화면에   Mary’s cosmetics   출력
- 문제2: 화면에  박씨가 소리질렀다. “도둑이야!”.  출력
- 문제3: 다음 코드 실행시켜서 \t와 \n의 역할 확인하기 <br>
print("안녕하세요.\n오늘\t날씨가\t\t좋습니다.")

In [ ]:
print("Mary's cosmetics")
print('박씨가 소리를 질렀다."도둑이야!!!!!!"')

print('박씨가 소리를 질렀다.\n"도둑이야!!!!!!"')


Mary's cosmetics
박씨가 소리를 질렀다."도둑이야!!!!!!"
박씨가 소리를 질렀다.
"도둑이야!!!!!!"
Mary's cosmetics
(박씨가 소리를 질렀다'."도둑이야!")


### 실습문제 :  구글TTS(gTTS) 사용하여 음성 출력하기

In [ ]:
# 구글 tts 프로그램
!pip install gTTS

In [ ]:
# 영어 출력
from gtts import gTTS
from IPython.display import Audio

eng_wav = gTTS('Hello World!')
eng_wav.save('eng.wav')

display(Audio('eng.wav', autoplay=True))

In [ ]:
# 한글 출력
from gtts import gTTS
from IPython.display import Audio

kor_wav = gTTS('안녕하세요! ', lang = 'ko')
kor_wav.save('kor.wav')

display(Audio('kor.wav', autoplay=True))

#gtts 사용법이라고 구글에서 검색해보기!!!

In [ ]:
# 한글 출력
from gtts import gTTS
from IPython.display import Audio

kor_wav = gTTS('kim soohwa! ', lang = 'ko')
kor_wav.save('kor.wav')

display(Audio('kor.wav', autoplay=True))

#gtts 사용법이라고 구글에서 검색해보기!!!

### 실습문제 :  파일 읽어 음성으로 출력하기

In [ ]:
# 파일 읽어서 음성으로 출력하기
from gtts import gTTS
from IPython.display import Audio

with open("./오감도.txt", 'r') as f:
    s = f.read()
    print(s)

kor_wav = gTTS(s, lang = 'ko')
kor_wav.save('kor.wav')

display(Audio('kor.wav', autoplay=True))



### **[문자열 인덱싱과 슬레이싱]**

In [ ]:
a = '우리는 민족중흥의 역사적 사명을 띠고 이 땅에 태어났다.'
print(a[3:12] )

fruits = 'orange'
print(fruits[1:3])

### 실습문제 : 문자열 거꾸로 출력하기

문자열을 입력 받아 거꾸로 출력하는 프로그램을 작성하세요

In [ ]:

a=input('문자열을 입력하세요.:')

 #a[:] 숫자를 안 넣으면 처음부터 끝까지 출력

print(f'거꾸로 출력:{a[::-1]}')


문자열을 입력하세요.:안녕
거꾸로 출력:녕안


### 실습문제 : 파일명.확장자명 분리하기

In [ ]:
a = input('확장자가 포함된 파일명을 입력하세요: ')

#split 함수 사용하여 작성하기




### **[변수 활용]**

#### 실습문제 : 지불할 음식 총 가격 계산하
- 음식가격:  4,450 원
- 세금: 6.75%
- 팁: 15%
1. meal 변수를 선언하고 값을 4450으로 지정하세요.
2. tax 변수를 생성하고 값으로 6.75%를 소수점 형태로 지정하세요.
3. tip 변수를 생성하고 값으로 15%를 소수점 형태로 지정하세요.
4. meal의 변수값에 세금을 곱해 meal 변수에 지정하세요.
5. 음식값에 tip을 더해 total 변수에 지정하세요.
6. total 변수 값을 출력하세요.

In [ ]:



meal = 4450
tax = 0.0675
tip = 0.15
meal = meal + (meal * tax)
total = meal + (meal * tip)
print(f'음식가격(세금포함): {meal}')
print(f'지출할 음식가격(팁포함): {round(total, 2)}')




음식가격(세금포함): 4750.375
지출할 음식가격(팁포함): 5462.93


In [ ]:
meal, tax, tip = 4450, 0.0675, 0.15
tax = meal * tax    # 세금으로 내야할 금액
meal = meal + tax   # 세금 포함된 음식가격
tip = meal * tip    # 팁으로 내야할 금액
total = meal + tip  # 내가 지불해야할 총 금액
print(f'세금으로 내야할 금액: {tax}')
print(f'세금 포함된 음식가격: {meal}')
print(f'팁으로 내야할 금액: {tip}')
print(f'내가 지불해야할 총 금액: {total}')

세금으로 내야할 금액: 300.375
세금 포함된 음식가격: 4750.375
팁으로 내야할 금액: 712.55625
내가 지불해야할 총 금액: 5462.93125


### 실습문제 : 섭씨온도를 화씨온도로 변환하기
- 섭씨 온도 (Celsius): 1 atm에서의 물의 어는점을 0도, 끓는점을 100도로 정한 온도 체계. (°C)
- 화씨 온도 (Fahrenheit): 물이 어는 온도는 32도, 물이 끓는 온도는 212도. 180등분. (°F)
- 섭씨온도(℃) = (5 / 9) × (℉ - 32)  # Celsius
- 화씨온도(℉) = (9 / 5) × ℃ + 32    # Fahrenheit

In [ ]:
input('섭씨 온도를 입력하세요:.')



거짓


--------

### **[연산자]**

#### 실습문제 :  동전 교환 프로그램
동전교환 프로그램을 작성하세요

In [ ]:
money=int(input('교환할 금액을 입력하세요:'))
print('='*50)

coin_500 =money //500
money%=500
print('500원짜리 ==>',coin_500,'개')

coin_100 =money //100
money%=100
print('100원짜리 ==>',coin_100,'개')

coin_50 =money //50
money%=50
print('50원짜리 ==>',coin_50, '개')

coin_10 =money //10
money%=10
print('10원짜리 ==>',coin_10, '개')

print('잔돈:',money,'원')







교환할 금액을 입력하세요:4321
500원짜리 ==> 8 개
100원짜리 ==> 3 개
50원짜리 ==> 0 개
10원짜리 ==> 2 개
잔돈: 1 원


In [ ]:
money=int(input('교환할 금액을 입력하세요:'))
print('='*50)

coin_500 =money //500

print('500원짜리 ==>',coin_500,'개')
money%=500

coin_100 =money //100
money%=100
print('100원짜리 ==>',coin_100,'개')

coin_50 =money //50
money%=50
print('50원짜리 ==>',coin_50, '개')

coin_10 =money //10
money%=10
print('10원짜리 ==>',coin_10, '개')

print('잔돈:',money,'원')


교환할 금액을 입력하세요:4321
500원짜리 ==> 8 개
100원짜리 ==> 3 개
50원짜리 ==> 0 개
10원짜리 ==> 2 개
잔돈: 1 원


In [ ]:
#강사님 코드

money = input('동전으로 교환할 금액을 입력하세요: ')
print('-'*50)
print(f'500원짜리    ==> {int(money)//500} 개')
money = int(money) % 500
print(f'100원짜리    ==> {money//100} 개')
money %= 100   # money = money % 100
print(f'50원짜리    ==> {money//50} 개')
money %= 50
print(f'10원짜리    ==> {money//10} 개')
money %= 10
print(f'잔돈        ==> {money} 원')

### **[리스트]**

- 리스트는 여러 개의 데이터 값을 하나의 변수에 담을 수 있는 데이터 구조
- 리스트는 순서(인덱스)가 있고 데이터를 수정, 삭제, 추가 가능하다.
- 대괄호로 표현되고 내부 원소는 ,(쉼표)로 구분됨

**리스트 생성 & 삭제**

In [ ]:
a = list()
a = []
a = [1,2,3]
a = ['I', 'Love', 'Python']
b = [1,2, '사랑해요', 'Python']
b = ['a', '1', 1, [2,3], -4]
a = [1,2,['a','b',['사랑해', '파이썬']], 3]

print(a)
print(b)



[1, 2, ['a', 'b', ['사랑해', '파이썬']], 3]
['a', '1', 1, [2, 3], -4]


In [ ]:
# 리스트 삭제
del(b)

print(a)
print(b)

#함수 삭제 후 출력할 수 없음

3


NameError: ignored

**리스트 요소에 접근**

In [ ]:
print( a )
print( a[1] )
print( a[2] )
print( a[2][1] )
# '파이썬'을 출력하려면?
print( a[2][2][1])
#index 2번의 2번째의 1번째



[1, 2, ['a', 'b', ['사랑해', '파이썬']], 3]
2
['a', 'b', ['사랑해', '파이썬']]
b
파이썬


**리스트 요소 추가 / 변경**

In [ ]:
a= ['가을','여름']
print(a)
a.append('봄')     # 리스트 맨 끝에 추가
print(a)
a.insert(1, '봄')  # 리스트명.insert(인덱스위치, 값),특정한 위치에 추가
print(a)
a.insert(3,'봄')
print(a)

['가을', '여름']
['가을', '여름', '봄']
['가을', '봄', '여름', '봄']
['가을', '봄', '여름', '봄', '봄']


**리스트 요소 삭제**

In [ ]:
print(a)
a.pop()     # 리스트 맨 끝 요소 삭제
print(a)
a.remove('봄')  # 리스트명.remove(값), 단 여러 개 일 경우 첫 번째만 삭제됨
print(a)

['가을', '봄', '여름', '봄', '봄']
['가을', '봄', '여름', '봄']
['가을', '여름', '봄']


### **[딕셔너리]**

**딕셔너리 키(key)와 값(value)에 접근하기**

In [ ]:
scores = {'국어': 100, '수학': 90, '영어': 80}

print(scores['국어'])  # 딕셔너리는 키로 접근한다.

# 방법1
print('---방법1---')
for key in scores:
    print('%s => %s' % (key, scores[key]))

# 방법2
print('---방법2---')
for key, value in scores.items():
    print('%s => %s' % (key, value))

100
---방법1---
국어 => 100
수학 => 90
영어 => 80
---방법2---
국어 => 100
수학 => 90
영어 => 80


In [ ]:
print(scores.keys())   # 키만 추출
print(scores.values()) # 키만 추출
print(scores.items())  # (키,값)쌍 형태로 추출

dict_keys(['국어', '수학', '영어'])
dict_values([100, 90, 80])
dict_items([('국어', 100), ('수학', 90), ('영어', 80)])


**딕셔너리에 요소 추가/수정/삭제**

In [ ]:
# 키로 요소 추가하기
scores['music'] = 100
print(scores)

# 요소 값 수정하기
scores['수학'] = 99
print(scores)

# 요소 삭제하기
del (scores['영어'])
print(scores)

{'국어': 100, '수학': 90, '영어': 80, 'music': 100}
{'국어': 100, '수학': 99, '영어': 80, 'music': 100}
{'국어': 100, '수학': 99, 'music': 100}


------

### 2-2. 조건문과 반목문으로 흐름 제어하기

### **[조건문]**

#### 실습문제 : 짝수 홀수 판별하기
숫자를 입력 받아 짝수/홀수를 판별하는 코드를 만들어 보세요.

In [ ]:
x = int(input('숫자를 입력해 주세요: '))

if x % 2 == 0:
    print('입력된 수 : %d ==> 짝수' % x)
else:
    print('입력된 수 : %d ==> 홀수' % x)

숫자를 입력해 주세요: 5
입력된 수 : 5 ==> 홀수


#### 실습문제 : 직각 삼각형 판별하기
피타고라스 정리에 의해 직각삼각형의 세 변의 길이에 대한 관계는 아래와 같습니다. 이 정리를 이용하여 세 변의 길이를 입력 받은 삼각형이 직각삼각형인지 아닌지 판별하는 프로그램을 만들어 보세요

In [ ]:
a=int(input('첫번째 변의 길이를 적으세요'))
b=int(input('두번째 변의 길이를 적으세요'))
c=int(input('세번째 변의 길이를 적으세요'))

length = [a,b,c]

if length [0]**2+length [1]**2 ==length [2]:
  print('직각삼각형 입니다')
else:
  print('직각삼각형이 아닙니다')


첫번째 변의 길이를 적으세요3
두번째 변의 길이를 적으세요4
세번째 변의 길이를 적으세요7
직각삼각형이 아닙니다


In [ ]:
#강사님 코드

### 실습문제 : 커피 자판기 조건문
아래 조건을 만족하는 커피 자동주문 머신을 만들어보세요.<br><br>
[커피 자동주문 머신]<br>
친구들과 현금 커피 자동주문 머신 앞에서
 - 아메리카노 2잔과 카페라테 2잔 카푸치노 1잔을 주문하였다.
 - 아메리카노 한잔에 2500원이고
 - 카페라테는 한잔에 3000원
 - 카푸치노는 한잔에 3000원이다.
커피 종류와 잔 수를 결정한 후 현금 15000원을 투입하고 잔돈을 1000원 받았다.

In [10]:
menu = '''[커피 자동주문 머신 메뉴]
------------------------------------------------------------
-아메리카노는 2500원
-카레라떼는 3000원
-카푸치노는 3000원
원하시는 커피종류와 잔수를 입력하세요.
------------------------------------------------------------'''
print(menu)

americano_price = 2500
cafelatte_price = 3000
cappuccino_price= 3000
a= int(input('아메리카노 주문 수량을 적어주세요.:'))
b=int(input('카페라떼 주문 수량을 적어주세요.:'))
c=int(input('카푸치노 주문 수량을 적어주세요.:'))

total_price=(americano_price*a)+(cafelatte_price*b)+(cappuccino_price*c)


print(f'지불할 총 금액은:{total_price}','원')

cash=int(input('돈을 넣어주세요.:'))

change=(cash)-(total_price)



while change < 0:
    print(f'결제금액이 {change}원 부족합니다.')
    minus = int(input('추가로 낸 금액을 적어주세요: '))
    cash += minus
    change = cash - total_price

print(f'잔돈: {change}원')

[커피 자동주문 머신 메뉴]
------------------------------------------------------------
-아메리카노는 2500원
-카레라떼는 3000원
-카푸치노는 3000원
원하시는 커피종류와 잔수를 입력하세요.
------------------------------------------------------------
아메리카노 주문 수량을 적어주세요.:2
카페라떼 주문 수량을 적어주세요.:2
카푸치노 주문 수량을 적어주세요.:1
총 합계:14000 원
돈을 적어주세요.:15000
잔돈: 1000원


In [ ]:
#강사님 코드
coffee1_price = 2500
coffee2_price = 3000
coffee3_price = 3000
menu = f'''[커피 자동주무 머신 메뉴]
--------------------------------
- 아메리카노 한잔에 {coffee1_price}원
- 카페라테는 한잔에 {coffee2_price}원
- 카푸치노는 한잔에 {coffee3_price}원
원하시는 커피 종류와 잔수를 입력하세요.
--------------------------------'''
print(menu)
coffee1 = int(input('아메리카노 몇잔: '))
coffee2 = int(input('카페라테   몇잔: '))
coffee3 = int(input('카푸치노   몇잔: '))
total = coffee1_price*coffee1 + coffee2_price*coffee2 \
                +coffee3_price*coffee3
print(f'지불할 총 금액은 {total} 원 입니다.')

money = int(input('돈을 넣어주세요: '))

if money < total:
    print(f'금액이 부족합니다. {total - money} 원을 더 넣어주세요.')
else:
    print(f'잔돈은 {money - total} 원 입니다.')
    print('커피 맛있게 드세요~~~~')

#### if와 else 문장을 --> elif 문장으로 변환하기

In [ ]:
# if, else 문장
pocket = ['paper', 'handphone']
card = True

if 'money' in pocket: #in 멤버쉽 연산자
    print('택시를 타고 가세요')
else:
    if card:
        print('택시를 타고 가세요')
    else:
        print('걸어서 가세요')

In [ ]:
# if elif 문장
pocket = ['paper', 'handphone']
card = True

if 'money' in pocket:
    print('택시를 타고 가세요')
elif card:
    print('택시를 타고 가세요')
else:
    print('걸어서 가세요')

#### 조건부 표현식

In [14]:
score = int(input('점수를 입력하세요: '))
message = "success!" if score >=60 else "failure!"
print(message)

점수를 입력하세요: 54
failure!


In [20]:
#짝수 홀수 판별하기 한줄 작성
a = int(input('정수를 입력하세요: '))
number='짝수'if a % 2==0 else "홀수"
print(number)


#강사님 코드
# 짝수 홀수 판별하기
x = int(input('판별할 숫자를 입력하세요: '))
print(f'숫자 {x}는 짝수!') if x%2==0 else  print(f'숫자 {x}는 홀수!')


정수를 입력하세요: 9
홀수


------

### **[반복문-for]**

#### 실습문제 : 1~N까지 합계 구하기
숫자를 입력받아 1~n까지의 합을 구할 때 짝수를 제외하고 합을 구하는 프로그램을 만들어 보세요.

In [30]:
## for 반복문
for 변수 in 유한개수(반복):
# 유한개수(반복)  사용할 수 있는 것들
range(시작, 종료, 증감분)함수
for x in range(0,3,1):  # 정수발생 range()함수 사용
    print(f'{x} 안녕하세요!')


IndentationError: ignored

In [ ]:
for i in range(0,3,1):
  print(f'{i} 안녕하세요')

for j in range(10,21,2):
  print(f'{j} 안녕하세요')

In [28]:
2. 문자열
a = '오렌지'
for x in a:   # 문자열 사용
    print(x)

사
과


In [ ]:
a=['사과','배']

In [ ]:
3. 리스트와 같은 유한 개 자료형
a = ['사과', '배', '포도', '샤인머스켓']
for x in a:   # 리스트 사용
    print(f'내가 좋아하는 과일은 {x}')

In [ ]:
# enumerate() : 인덱스 정보를 준다.
a = ['사과', '배', '포도', '샤인머스켓']
for idx, x in enumerate(a):   # 리스트 사용
    print(f'{idx} 내가 좋아하는 과일은 {x}')

In [39]:
n=int(input('양의 정수n을 입력하세요:'))

hap=0

for a in range(1,n+1):
    if a % 2 ==1:
      hap= hap+a


print(f"1부터 {n}까지 짝수를 제외한 합은 {hap} 입니다.")





양의 정수n을 입력하세요:3
1부터 3까지 짝수를 제외한 합은 4 입니다.


In [36]:
#강사님 코드
n=int(input('양의 정수 n을 입력해주세요'))
hap=0

for i in range(1,n+1,1):
  hap=hap+i


print(f"1부터 {n}까지 짝수를 제외한 합은 {hap} 입니다.")

양의 정수 n을 입력해주세요3
1부터 3까지 짝수를 제외한 합은 6 입니다.


### [실습문제] turtle로 정오각형 그리기
#### 주의! : turtle은 가능한 IDLE 또는 에디터에서 사용한다.
Turtle 이용해서 정오각형을 그리는 코드이다. 이것을 for 반복문을 이용하여 코드를 줄여보세요.

In [ ]:
import turtle

t=turtle.Pen()    # 펜블럭
# t.shape('turtle') # 커서 모양
# t.color('red')    # 선 색생
# t.pensize(5)      # 선 굵기

t.forward(100)
t.right(72)
t.forward(100)
t.right(72)
t.forward(100)
t.right(72)
t.forward(100)
t.right(72)
t.forward(100)
t.right(72)

turtle.exitonclick()

In [ ]:
import turtle

t=turtle.Pen()    # 펜블럭

t.shape('turtle') # 커서 모양
t.color('blue')    # 선 색생
t.pensize(5)      # 선 굵기

for i in range(5):
    t.forward(100)
    t.right(360/5)

turtle.exitonclick()

In [ ]:
#벌집 그리기
import turtle

t=turtle.Turtle()
t.up() #펜 올리기
t.goto(-30, 50)#좌표로 이동하기
t.down()#펜 내리기

def honeycomb(): # 벌집 그리기
    for i in range(6):
        for _ in range(6):
            t.forward(100)
            t.left(60)
        t.forward(100)
        t.right(60)

honeycomb()
turtle.exitonclick()

In [ ]:
# 마우스로 클릭한 위치에 별 그리기
import turtle
t = turtle.Turtle()

t.speed(0)  # 0: 그리기 속도 가장 빠르게
t.screen.bgcolor('black')
t.color('yellow') # (거북이)그리기 색상 노랑
t.clear()         # 화면 지우기
t.penup()

# 별 그리기
def drawStar(x, y):
    t.goto(x, y)
    t.pendown()
    t.begin_fill()
    for _ in range(5):
        t.forward(20)
        t.right(144)
    t.end_fill()
    t.penup()

# 마우스로 화면 클릭하기
turtle.onscreenclick(drawStar)
t.ht()

# 화면에 포커스가 있는 상태 유지하기
turtle.done()

In [40]:
!pip install ColabTurtle

  Preparing metadata (setup.py) ... done
  Created wheel for ColabTurtle: filename=ColabTurtle-2.1.0-py3-none-any.whl size=7641 sha256=b498c743806c42ab82594f5c614a02f46f4c6a520579e8e49d097117bcf5ec09
  Stored in directory: /root/.cache/pip/wheels/5b/86/e8/54f5c8c853606e3a3060bb2e60363cbed632374a12e0f33ffc
Successfully built ColabTurtle


In [ ]:
import ColabTurtle.turtle
t = turtle.Turtle()

t.speed(0)  # 0: 그리기 속도 가장 빠르게
t.screen.bgcolor('black')
t.color('yellow') # (거북이)그리기 색상 노랑
t.clear()         # 화면 지우기
t.penup()

# 별 그리기
def drawStar(x, y):
    t.goto(x, y)
    t.pendown()
    t.begin_fill()
    for _ in range(5):
        t.forward(20)
        t.right(144)
    t.end_fill()
    t.penup()

# 마우스로 화면 클릭하기
turtle.onscreenclick(drawStar)
t.ht()

# 화면에 포커스가 있는 상태 유지하기
turtle.done()

### **[반복문-while]**

In [ ]:
num1, num2 = 0, 0

while True:
    num1 = int(input("더할 첫 번째 숫자: "))
    if num1 == 0:
        break
    num2 = int(input("더할 두 번째 숫자: "))

    print(f'{num1} + {num2} = {num1 + num2}')

print("0을 입력해서 반복문을 탈출했습니다.")


더할 첫 번째 숫자: 10
더할 두 번째 숫자: 20
10 + 20 = 30
더할 첫 번째 숫자: 55
더할 두 번째 숫자: 30
55 + 30 = 85
더할 첫 번째 숫자: 0
0을 입력해서 반복문을 탈출했습니다.


### 실습문제 : 나무 열 번 찍어 넘기기

In [43]:

treehit = 0

while treehit < 10:
   print(f"나무를 {treehit}번 찍었습니다.")
   treehit += 1
    if treehit == 10:
        print("나무가 넘어갑니다!")



IndentationError: ignored

In [50]:
treehit = 0

while treehit < 10:
  treehit += 1
  print(f"나무를 {treehit}번 찍었습니다.")
  if treehit == 10:
    print("나무가 넘어갑니다!~~~~")

나무를 1번 찍었습니다.
나무를 2번 찍었습니다.
나무를 3번 찍었습니다.
나무를 4번 찍었습니다.
나무를 5번 찍었습니다.
나무를 6번 찍었습니다.
나무를 7번 찍었습니다.
나무를 8번 찍었습니다.
나무를 9번 찍었습니다.
나무를 10번 찍었습니다.
나무가 넘어갑니다!~~~~


#### 실습문제: 무한 반복 빠져나오기
무한 반복문을 빠져나오기 위한 프로그램을 만들어 보세요

In [ ]:
prompt = """
    1.Add   2.Del   3.Modify   4.Quit

    Enter number: """   # 1.코드를 실행하면 출력되는 부분

number = 0






#### 실습문제
공을 일정 높이에서 던졌을 때, 원래 높이의 1/2만큼 튀어 오른다고 합니다. 그러다가 높이가 0.00001m보다 낮으면 튀어오르지 않는다고 합니다. while문을 이용하여 이 프로그램을 만들어 보세요.

In [60]:
length = int(input('공의 높이를 입력하세요(meter) : '))
cnt, meter = 0, length
while True:
    cnt += 1    # 바닦에 닿았다.
    meter /= 2  # 길이가 1/2으로 줄어든다.
    if meter < 0.00001:
        break
print(f'{length} 위치에서 떨어진 공은 {cnt} 번 튕겼습니다.')


공의 높이를 입력하세요(meter) : 50
50위치에서 공의 튕긴 횟수는 총 23번 입니다.


In [ ]:
length = int(input('공의 높이를 입력하세요(meter) : '))
cnt, meter = 0, length
while meter >= 0.00001:
    cnt += 1    # 바닦에 닿았다.
    meter /= 2  # 길이가 1/2으로 줄어든다.
print(f'{length} 위치에서 떨어진 공은 {cnt} 번 튕겼습니다.')

#### 실습문제 : 도둑잡기 게임 만들기
경찰에 쫓기던 범인이 3개의 방 중에 한 곳에 숨어버렸습니다. 범인은 방문을 열 때마다 무작위로 3개의 방 중 하나의 방으로 이동합니다. 3개의 방 중 하나의 방을 선택했을 때 범인이 있다면 100점, 없을 때 -10점을 부여하고 범인을 잡으면 게임을 종료하는 프로그램을 만들어 보세요.

In [64]:

import random
score = 0

while True:  # 무한반복
    # 1.무작위로 범인 방 배치
  room=random.randint(1,3)


    # 2.사용자로부터 방 번호 입력 받기
  choice_number=int(input'방 번호를 선택하세요,(1,2,3):')



    # 3.사용자가 입력한 방 번호와 범인이 있는 방 비교
  if room==choice_number:
    print("범인체포")
    score+=100
    bresk
  else:
    print('범인이 있는 방이 아닙니다')
    score-=10


print('게임 종료!')        # 게임 종료 메시지 출력
print(f'점수: {score} 점') # 최종 점수 출력

SyntaxError: ignored

In [ ]:
#강사님 코드
mport random
score = 0

while True:  # 무한반복
    # 1.무작위로 범인 방 배치
    room = random.randint(1,3)  # 1~3번 중 임의의 정수

    # 2.사용자로부터 방 번호 입력 받기
    n = int(input('방 번호를 입력하세요: '))

    # 3.사용자가 입력한 방 번호와 범인이 있는 방 비교
    if room == n:
        print('범인 체포!!!')
        score += 100
        break
    else:
        print('범인이 있는 방이 아닙니다. 다시 선택하세요.')
        score -= 10
    print('---------------')

print('게임 종료!')        # 게임 종료 메시지 출력
print(f'점수: {score} 점') # 최종 점수 출력



### 실습문제 :  커피 자동주문 프로그램 업그레이드
앞에서 만들어 본 커피 자동주문 머신 프로그램을 while문을 이용하여 돈을 모두 지불할 때까지 돈을 입력 받도록 프로그램 기능 수정하기

In [67]:
#강사님 코드
coffee1_price = 2500
coffee2_price = 3000
coffee3_price = 3000
menu = f'''[커피 자동주무 머신 메뉴]
--------------------------------
- 아메리카노 한잔에 {coffee1_price}원
- 카페라테는 한잔에 {coffee2_price}원
- 카푸치노는 한잔에 {coffee3_price}원
원하시는 커피 종류와 잔수를 입력하세요.
--------------------------------'''
print(menu)
coffee1 = int(input('아메리카노 몇잔: '))
coffee2 = int(input('카페라테   몇잔: '))
coffee3 = int(input('카푸치노   몇잔: '))
total = coffee1_price*coffee1 + coffee2_price*coffee2 \
                +coffee3_price*coffee3
print(f'지불할 총 금액은 {total} 원 입니다.')

while True:
    print('-' * 50)
    money = int(input('돈을 넣어주세요: '))

    change = total - money  # 필요한 금액

    if change > 0:
        print(f'금액이 부족합니다. {change} 원을 더 넣어주세요.')
        total = change
    else:
        print(f'잔돈은 {abs(change)} 원 입니다.')
        print('커피 맛있게 드세요~~~~')
        break
print('----------------끝!')


[커피 자동주무 머신 메뉴]
--------------------------------
- 아메리카노 한잔에 2500원
- 카페라테는 한잔에 3000원
- 카푸치노는 한잔에 3000원
원하시는 커피 종류와 잔수를 입력하세요.
--------------------------------
아메리카노 몇잔: 2
카페라테   몇잔: 2
카푸치노   몇잔: 1
지불할 총 금액은 14000 원 입니다.
--------------------------------------------------
돈을 넣어주세요: 10000
커피 맛있게 드세요


NameError: ignored

In [68]:
menu = '''[커피 자동주문 머신 메뉴]
------------------------------------------------------------
-아메리카노는 2500원
-카레라떼는 3000원
-카푸치노는 3000원
원하시는 커피종류와 잔수를 입력하세요.
------------------------------------------------------------'''
print(menu)

americano_price = 2500
cafelatte_price = 3000
cappuccino_price= 3000
a= int(input('아메리카노 주문 수량을 적어주세요.:'))
b=int(input('카페라떼 주문 수량을 적어주세요.:'))
c=int(input('카푸치노 주문 수량을 적어주세요.:'))

total_price=(americano_price*a)+(cafelatte_price*b)+(cappuccino_price*c)


print(f'지불할 총 금액은:{total_price}','원')

cash=int(input('돈을 넣어주세요.:'))

change=(cash)-(total_price)



while change < 0:
    print(f'결제금액이 {change}원 부족합니다.')
    minus = int(input('추가로 낸 금액을 적어주세요: '))
    cash += minus
    change = cash - total_price

print(f'잔돈: {change}원')

[커피 자동주문 머신 메뉴]
------------------------------------------------------------
-아메리카노는 2500원
-카레라떼는 3000원
-카푸치노는 3000원
원하시는 커피종류와 잔수를 입력하세요.
------------------------------------------------------------
아메리카노 주문 수량을 적어주세요.:2
카페라떼 주문 수량을 적어주세요.:2
카푸치노 주문 수량을 적어주세요.:1
지불할 총 금액은:14000 원
돈을 넣어주세요.:10000
결제금액이 -4000원 부족합니다.
추가로 낸 금액을 적어주세요: 8000
잔돈: 4000원


### **[난수(Random Number)]**

### 실습문제 : 임의의 두 수 합 구하기
1부터 10까지의 범위를 갖는 임의의 두 정수 a, b를 더하여 결과 출력하기

In [71]:
import random

a = random.randint(1, 10)
b = random.randint(1, 10)
result =a+b
print(f' a의 랜덤 숫자:{a}')
print(f' b의 랜덤 숫자:{b}')
print(f'{a}+{b}={result}입니다:')


 a의 랜덤 숫자:9
 b의 랜덤 숫자:6
9+6=15입니다:


### 실습문제 : 던져 나온 주사위의 눈금별 횟수 카운팅
주사위를 100번 던져서 나오는 숫자를 각 숫자별로 카운트하고 가장 많이 나온 숫자와 그 숫자의 횟수를 출력하시오.

In [80]:
import random
dice = [0,0,0,0,0,0]
for i in range(100):
    x = random.randint(1,6)
    dice[x-1] =  dice[x-1] + 1
print(f'주사위 숫자별 나온 횟수: {dice}')
print(f'가장 많이 나온 횟수는: {max(dice)}')
print(f'가장 많이 나온 횟수의 주사위 눈금은: { dice.index(max(dice))+1}')


주사위 숫자별 나온 횟수: [17, 21, 11, 17, 21, 13]
가장 많이 나온 횟수는: 21
가장 많이 나온 횟수의 주사위 눈금은: 2


In [ ]:
# 성적 :
score = [90, 100, 80, 99, 88]
student_score = []
for x in score:
    student_score.append(f'수학 {x}점')
print(student_score)

#줄여서 사용함

score = [90, 100, 80, 99, 88]
student_score = [ f'수학 {x}점' for x in score ]
print(student_score)

------

### 2-3. 함수를 이용하여 알고리즘 만들기

### **[함수 & 모듈]**

In [ ]:
def nSum(a, b):
    c = a + b
    return c

nSum(50, 3)

####  함수의 구조
매개변수와 리턴값에 따른 함수의 유형 예

- 매개변수(X), 리턴값(X)

In [ ]:
def func():
    print('안녕하세요!')

print( func() )

- 매개변수(X), 리턴값(O)

In [ ]:
def func():
    print('안녕하세요!')
    return '처리완료!'

print( func() )

- 매개변수(O), 리턴값(X)

In [ ]:
def func(a, b):
    print('덧셈: %d + %d = %d' % (a, b, a+b))

print( func(5, 7) )

- 매개변수(O), 리턴값(O)

In [ ]:
def func(a, b):
    print('덧셈: %d + %d = %d' % (a, b, a+b))
    return a+b

print( func(5, 7) )

### 실습문제 : 암스트롱수 구하기
다음 조건을 참고하여 암스트롱수를 구하는 함수를 만들어 출력하는 프로그램을 만들어 보세요.
- **암스트롱수** : 세 자리의 정수 중에서 각 자리의 수를 세 제곱한 수의 합과 자신이 같은 수

In [ ]:
# 암스트롱수 구하기
def is_armstringNumber(N):
    N100 = N//100
    N10  = (N - N100*100)//10
    N1   = (N - N100*100 - N10*10)
    # print(f'{N100} {N10} {N1}')
    if N == (N100**3 + N10**3 + N1**3):
        return True
    else:
        return False

for i in range(100, 1000):
    if is_armstringNumber(i):
        print(f'[{i}] 암수트롱 수!')




#### 실습문제
아래 조건을 만족하는 커피 자동주문 머신을 만들어보세요.

IndentationError: ignored

### 2-4. [미션] 응용 프로그램 만들기

#### 실습문제
아래 조건을 만족시키는 <영어단어장 만들기> 프로그램을 만들어 보세요.

영단어는 key, 그의 뜻은 value 로 두는 dictionary 를 만드세요.<br>
<메뉴>
 - 1. 단어 등록
 - 2. 단어 검색
 - 3. 단어 삭제
 - 4. 모든 단어 보기
 - 0. 종료하기

1. 단어 등록 <br>
    새 영단어와 그의 뜻을 입력 받고 dictionary 에 추가합니다.

2. 단어 검색<br>
    영단어를 입력 받아 그의 뜻을 출력합니다.<br>
    미등록된 단어인 경우 '미등록 단어'를 출력합니다.
    
3. 단어 삭제<br>
    영단어를 입력 받아 단어장에서 해당 단어를 삭제한다.<br>
    미등록된 단어인 경우 '미등록 단어'를 출력합니다
    
4. 모든 단어 보기<br>
    현재 등록되어 있는 모든 단어와 뜻을 출력합니다.

0. 종료하기



---
THE END



------